enderio-machines/src/main/java/com/enderio/machines/common/blockentity/XPVacuumBlockEntity.java

```diff
@@ -0,0 +1,128 @@
+package com.enderio.machines.common.blockentity;
+
+import java.util.List;
+
+import com.enderio.base.common.blockentity.sync.FluidStackDataSlot;
+import com.enderio.base.common.blockentity.sync.IntegerDataSlot;
+import com.enderio.base.common.blockentity.sync.SyncMode;
+import com.enderio.machines.common.MachineTier;
+import com.enderio.machines.common.blockentity.base.MachineBlockEntity;
+import com.enderio.machines.common.blockentity.data.sidecontrol.fluid.FluidTankMaster;
+import com.enderio.machines.common.menu.XPVacuumMenu;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.world.entity.ExperienceOrb;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.material.Fluids;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.IFluidHandler.FluidAction;
+
+public class XPVacuumBlockEntity extends MachineBlockEntity {
+	private static final double COLLISION_DISTANCE_SQ = 1 * 1;
+    private static final double SPEED = 0.025;
+    private static final double SPEED_4 = SPEED ;
+	private static final int MAX_RANGE = 6;
+	private int range = 6;
+	private FluidTankMaster fluidTank;
+
+	public XPVacuumBlockEntity(BlockEntityType<?> pType, BlockPos pWorldPosition,
+			BlockState pBlockState) {
+		super(MachineTier.STANDARD, pType, pWorldPosition, pBlockState);
+		this.fluidTank =  new FluidTankMaster(Integer.MAX_VALUE, getIoConfig()); //that seems quite large?
+		addDataSlot(new FluidStackDataSlot(() -> fluidTank.getFluidInTank(0), fluidTank::setFluid, SyncMode.WORLD));
```

Comment:
as only one fluid should be here, maybe think about using an in integer dataslot as the fluid isn't required to be synced

Reply:
Good idea, hadn't considered that.